src/config.js

```diff
@@ -0,0 +1,56 @@
+/* @flow */
+import requireUncached from 'require-uncached';
+
+import {createLogger} from './util/logger';
+import {UsageError} from './errors';
+
+const log = createLogger(__filename);
+
+type ApplyConfigToArgvParams = {|
+  argv: Object,
+  configObject: Object,
+  defaultValues: Object,
+|};
+
+export function applyConfigToArgv({
+  argv,
+  configObject,
+  defaultValues = {},
+}: ApplyConfigToArgvParams): Object {
+  const newArgv = {...argv};
+  for (const option in configObject) {
+    const wasValueSetOnCLI = typeof(newArgv[option]) !== 'undefined' &&
```

Comment:
When someone reads this code, it won't be immediately clear what's going on since we both know it's a hack. For situations like this, you should add a comment above it explaining in plain English why you are assuming the value was set on the command line.